core/consensus/aura/src/lib.rs

```diff
@@ -214,10 +215,11 @@ pub fn start_aura_thread<B, C, E, I, P, SO, Error, OnExit>(
 }
 
 /// Start the aura worker. The returned future should be run in a tokio runtime.
-pub fn start_aura<B, C, E, I, P, SO, Error, OnExit>(
+pub fn start_aura<B, C, SC, E, I, P, SO, Error, OnExit>(
 	slot_duration: SlotDuration,
 	local_key: Arc<P>,
 	client: Arc<C>,
+	select_chain: SC,
```

Comment:
unfortunately, this breaks API. We will have to find a way to do this that doesn't.

Reply:
Yes, this change will inevitably break the api - as it needs to be specified and at the service-intend-layer and somehow pushed down to the authoring and finalisation. Even if we tried to still have client facilitate this somehow, this would mean adding the same `SC` on every `Where C: Client<...SC>`, which would also break this same call.

I am afraid there is just no way around this breaking the API.

Reply:
The only thing we _could_  do is have client implement the new trait as well and hard-code that to the `LongestChain` implementation and directly deprecate its usage. But then still, would the where clause later on have to change to the new `trait` and thus, this would still break the API.

Reply:
This is currently the blocker for me on this PR. I am not sure how we even can resolve this without creating a huge other mess elsewhere.

The _only_ way I can currently think of is by having a compatibility layer, that is `Client` implements `LongestChain` `SC`, however in that case, a) the where clause still needs to be extended (and that is "breaking the API", as it now constrains more than it was before) and b) it would mean this function doesn't actually allow any other implementation and thus isn't really implementing the ticket (or solving our problem).

Not sure what to do about this?!?